src/twisted/newsfragments/7530.bugfix

```diff
@@ -0,0 +1 @@
+Allow for key and cert .pem files that do not have a a trailing newline
```

Comment:
This is slightly inaccurate because `ssl.PrivateCertificate.loadPEM` still requires them; the requirement is just omitted in one place (`ssl:` endpoints) and not everywhere.

Perhaps you should just fix this issue inside `loadPEM` so that it doesn't have to be a more complex and restricted NEWS entry?